internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -436,6 +436,7 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 		if ok {
 			cd.SetNamespace(or.Resource.GetNamespace())
 			cd.SetName(or.Resource.GetName())
+			cd.SetGenerateName(or.Resource.GetGenerateName())
```

Comment:
Is this intentional? Doesn't look related for the rest of the PR 🤔

Reply:
i see a new e2e test check for this behavior in `TestBasicCompositionCluster`, but I don't see it mentioned in the PR description or a mention of a tracking issue. Can you help link this up for us @jcogilvie?